visiont/scripts/__main__.py

```diff
@@ -12,10 +13,66 @@ def main():
 
     Formatter = argparse.ArgumentDefaultsHelpFormatter
 
-    train = subcmd.add_parser("train", help="trains the vision transformer", formatter_class=Formatter)
-    train.add_argument("dataset", type=Path, help="path to ImageNet-like dataset directory")
-    train.set_defaults(main=visiont.scripts.train.main)
+    train = subcmd.add_parser(
+        "train", help="trains the vision transformer", formatter_class=Formatter
+    )
+    val = subcmd.add_parser(
+        "val",
+        help="validates the trained vision transformer",
+        formatter_class=Formatter,
+    )
+    data_gen = subcmd.add_parser(
+        "generate",
+        help="generates data to see training sample diversity",
+        formatter_class=Formatter,
+    )
+    train.add_argument(
+        "-d",
+        "--dataset",
+        dest="dataset",
```

Comment:
We don't need `dest` here and below. With `--dataset` the arg will get stored in `args.dataset` already :hugs:

Reply:
But also: the `--cmd` flags need `required=True` if you want them to be required (like dataset e.g.) otherwise `args.dataset` can be `None`